internal/promotion/runner/builtin/git_cloner.go

```diff
@@ -103,10 +103,24 @@ func (g *gitCloner) run(
 			SSHPrivateKey: creds.SSHPrivateKey,
 		}
 	}
+	var repoUser git.User
+
+	if cfg.User != nil {
+		repoUser = git.User{
+			Name:  cfg.User.Name,
+			Email: cfg.User.Email,
+			SigningKey:     cfg.User.SigningKey, // Optional, may be empty
+			SigningKeyType: g.gitUser.SigningKeyType, // Preserve system-level SigningKeyType
```

Comment:
This can be hard-coded or, even better, omitted entirely with the assumption of #3925 being merged before this PR. (It will be, because other changes in this PR depend on it.) Starting with #3925, GPG will be assumed when key type is not specified.

Reply:
```suggestion
```